### PR TITLE
[MODFISTO-400-2]-updated schema for tenant and user_tenant table .

### PR DESCRIPTION
**Pull request by SinghAdes**
## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all the appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?
- Did you modify code to call some additional endpoints?
  - [ ] If so, do you check that necessary module permission added in ModuleDescriptor-template.yaml file?

Ideally, all the PRs involved in breaking changes would be merged on the same day
to avoid breaking the folio-testing environment.
Communication is paramount if that is to be achieved,
especially as the number of inter-module and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems,
ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/resources/db/changelog/changes/create-tenant-table.xml [hunk 1]
```diff
@@ -7,7 +7,7 @@
   <changeSet id="MODFISTO-394@@create-tenant-table" author="singhadesh">
     <createTable tableName="tenant">
 
-      <column name="tenant_id" type="text">
+      <column name="id" type="text">
         <constraints primaryKey="true" primaryKeyName="pk_tenant_id" nullable="false"/>
       </column>
```
Comment (SerhiiNosko) [inline src/main/resources/db/changelog/changes/create-tenant-table.xml:13]: Please also rename tenant_name -> name

File: src/main/resources/db/changelog/changes/create-user-tenant-table.xml [hunk 2]
```diff
@@ -7,15 +7,18 @@
   <changeSet id="MODFISTO-400@@create-user-tenant-table" author="azizbekxm">
     <createTable tableName="user_tenant">
 
-      <column name="user_id" type="uuid">
+      <column name="id" type="uuid">
         <constraints primaryKey="true" unique="true" nullable="false"/>
       </column>
+      <column name="user_id" type="uuid">
+        <constraints unique="false" nullable="false"/>
```
Comment (SerhiiNosko) [inline src/main/resources/db/changelog/changes/create-user-tenant-table.xml:14]: user_id always will be unique, so please change to unique="true"

**Review by SerhiiNosko — APPROVED** [review]

**Comment by SinghAdes** [comment]
image
image

**Comment by damien-git** [comment]
@SinghAdes If we're going to search user_tenant by username, maybe we should define an index ?

**Review by Abdulkhakimov — APPROVED** [review]

File: src/main/resources/db/changelog/changes/create-user-tenant-table.xml [hunk 3]
```diff
@@ -5,23 +5,33 @@
...
       <column name="is_primary" type="boolean" defaultValueBoolean="false">
         <constraints unique="false" nullable="false"/>
       </column>
-
     </createTable>
+
+    <createIndex indexName="user_id_idx" tableName="user_tenant">
+      <column name="user_id"/>
+    </createIndex>
+    <createIndex indexName="user_name_idx" tableName="user_tenant">
```
Comment (SerhiiNosko) [inline src/main/resources/db/changelog/changes/create-user-tenant-table.xml:31]: lets make name consistent with fieldname, so name shoud be username_idx I suppose

**Pull request merged**